### PR TITLE
Allowing publisher toggle to include a site outside of the minimum payment requirements

### LINKED ISSUE #13879: URL Heart (publisher toggle) button should be able to include sites outside of minimum requirements
### Description
Previously discussed with @NejcZdovc & @bsclifton, related to: #12675 

Currently, the publisher toggle button in the navigation bar (heart) functions only as an inclusion toggle, that is:

It does not add a site to the ledger if it is not already in it, it only can pin and unpin sites that are already in the ledger.

The suggestion is that it is able to include sites from the ledger entirely. For example:

Navigating to news.ycombinator.com, with minimum inclusion times set 1 minute long with 5 visits, a user should be able after only a few seconds have passed, to click the publisher toggle and see ycombinator.com in the ledger 5 minutes total time, and 5 visits.

From that point on, the publisher toggle will then retain its current functionality of acting as a pinned state toggle. Users should navigate to about:preferences#payments if they wish to remove the site from payments.

It seems natural that this button drives control of what the user would like to see in the ledger.

### Additional Information
Currently the button has three states, the language of them here:
1. **Site not present in the ledger** -> _"Publisher is not yet added to the ledger, because it doesn't meet criteria yet"_ (No action taken on click, suggested feature would include publisher in the ledger)
2. **Site present in ledger, pinned** -> _"This publisher is enabled for payment, click to disable"_ (Site unpinned on click)
3. **Site present in ledger, unpinned** -> _"Click to enable this publisher for payments"_ (Site pinned on click)

cc: @bradleyrichter for feedback about the issue in general, as well as the new language for the hover states

**Note: This has a cross dependency with #13856**
The code for a manual addition outside of set minimum requirements will need to be leveraged for this feature

If viewing this functionality is desired, it can be viewed in [this branch](https://github.com/ryanml/browser-laptop/tree/ledger-heart-fix) (This would be rebased with #13856 once it gets merged)

### PR DESCRIPTION
**Pull request by ryanml**
Fixes #13879 

Simply put, this allows the publisher toggle button to act as a mechanism of inclusion for a site that does not yet meet the minimum criteria.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
**Test plan will update as work progresses**
1. Enable Brave Payments
2. Set minimums rather high (Ex: 1 minute visit time with 5 minimum visits)
3. Navigate to a web page.
4. Click on the Publisher Toggle button.
5. Confirm that the button 'enables', and that the web page has been added to the ledger with the minimum criteria.
6. Confirm that the site can be unpinned/pinned by further interacting with the button.

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by ryanml** [comment]
Functionality implemented, to do:

1. Add additional tests
2. Update Publisher Toggle tooltips

File: js/actions/appActions.js [hunk 1]
```diff
@@ -1834,10 +1834,11 @@ const appActions = {
    * Dispatches a message to add a given publisher to the ledger.
    * @param location - the URL of the publisher
    */
-  addPublisherToLedger: function (location) {
+  addPublisherToLedger: function (location, tabId = false) {
```
Comment (NejcZdovc) [inline js/actions/appActions.js]: This shouldn't be false, where we are expecting number for this param. I think default should be `tabState.TAB_ID_NONE`

**Comment by NejcZdovc** [comment]
master 9a04a69
0.23 6045445ad
0.22 f6536e6a0

**Comment by btlechowski** [comment]
The formula for time spent on the site is: minimum page time * minimum number of visits.
It seems to be done by design.

Verified on Ubuntu 17.10 x64
0.22.703 903b8d0d
libchromiumcontent 66.0.3359.139
muon: 6.0.8

**Pull request merged**